code/mob.dm

```diff
@@ -154,6 +154,8 @@
 	var/speech_void = 0
 	var/now_pushing = null //temp. var used for Bump()
 	var/atom/movable/pushing = null //Keep track of something we may be pushing for speed reductions (GC Woes)
+	var/singing = 0
```

Comment:
are you reusing this var to hold both numbers and the singing strings?

Reply:
The values will be 0 for no singing, 1 for normal singing, "bad", or "loud"

Reply:
To make it clearer I will change it so that no singing is represented with an empty string and normal singing with "normal"